### PR TITLE
Blog Post: TypeScript and GraphQL union 

### PR DESCRIPTION
**Pull request by TuvalSimha**

### CODE REVIEW

File: pages/blog/codegen-this_is_my_type.mdx [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+  tags: ['graphql', 'typescript', 'codegen'],
+  date: '2022-05-23',
+  description: 'How to create TypeScript Discriminated Union with GraphQL Code Generator',
+  image: '/blog-assets/graphql-code-generator/cover.png'
+}
+export default withArticle(meta)
+ 
+# Introduction
+
+In the last few months i just started learn how to code, i already know how to partly work with JavaScript and start to learn TypeScript. When i was create TypeScript types i found a lot if tutorials and documentation that help me create Types and how to implement them in my code. For example: **Discriminated Union.**
```
Comment (B2o5T) [inline pages/blog/codegen-this_is_my_type.mdx:15]: I must be always uppercased ;)
```suggestion
In the last few months I just started learn how to code, I already know how to partly work with JavaScript and start to learn TypeScript. When I was create TypeScript types I found a lot of tutorials and documentation that help me create Types and how to implement them in my code. For example: **Discriminated Union.**
```

File: pages/blog/codegen-this_is_my_type.mdx [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+
+If you ask yourself "Hey, what it is discriminated union?", Lets talk about it!
```
Comment (B2o5T) [inline pages/blog/codegen-this_is_my_type.mdx:17]:
```suggestion
If you ask yourself "Hey, what is discriminated union?", let's talk about it!
```

File: pages/blog/codegen-this_is_my_type.mdx [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+
+So a discriminated union is kind of pattern, that tells the compiler all possible types that can be used in a variable.
+If you want to create discriminated union, all types must have at least one unique value that can be used to identify the type.
+In addition, you must have same literal values for all types (For example: string, number and boolean).
```
Comment (B2o5T) [inline pages/blog/codegen-this_is_my_type.mdx:21]:
```suggestion
So a discriminated union is a kind of pattern, that tells the compiler all possible types that can be used in a variable.
If you want to create discriminated union, all types must have at least one unique value that can be used to identify the type.
In addition, you must have the same literal values for all types (For example: string, number and boolean).
```

File: pages/blog/codegen-this_is_my_type.mdx [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+
+## Lets split it to 3 simple parts:
+
+1. Discriminant - Where i create the Types.
+2. Union - Take the types i want and create union type for them.
+3. How to use (Type Guards) - Use my new types in my code.
```
Comment (B2o5T) [inline pages/blog/codegen-this_is_my_type.mdx:27]:
```suggestion
1. Discriminant - Where I create the Types.
2. Union - Take the types I want and create a union type for them.
3. How to use (Type Guards) - Use my new types in my code.
```

File: pages/blog/codegen-this_is_my_type.mdx [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+}
+```
+
+
+## GraphQL Codegen plugin - Types Safety:
+
+* This why of work with Types is great, but very dangerous.
+Imagine you should create a project with a lot of different Types, there is a chance you probably will make mistakes with the Types in code-time.
+
+* You need to mange the Types according the data base, the server protocol and inside the Types file.
```
Comment (B2o5T) [inline pages/blog/codegen-this_is_my_type.mdx:81]:
```suggestion
* This why of work with Types is great, but very dangerous.
Imagine you should create a project with a lot of different Types, there is a chance you probably will make mistakes with the Types in code-time.

* You need to manage the Types according to the database, the server protocol and inside the Types file. 
```

File: pages/blog/codegen-this_is_my_type.mdx [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+* This why of work with Types is great, but very dangerous.
+Imagine you should create a project with a lot of different Types, there is a chance you probably will make mistakes with the Types in code-time.
+
+* You need to mange the Types according the data base, the server protocol and inside the Types file. 
+
+## Summary:
+
+So after i learned about TypeScript types (And did a lot if mistakes in code-time), just figured out that GraphQL Codegen make my work easier, faster, and accurate.
+
+Codegen helps you create TypeScript Types based on your GraphQL schema. This way of working can help you reduces the chances to write Types mistakes, and probably find the bugs at build-time and not running-time.
```
Comment (B2o5T) [inline pages/blog/codegen-this_is_my_type.mdx:87]:
```suggestion
So after I learned about TypeScript types (And did a lot of mistakes in code-time), just figured out that GraphQL Codegen makes my work easier, faster, and more accurate.

Codegen helps you create TypeScript Types based on your GraphQL schema. This way of working can help you reduces the chances to write Types mistakes, and probably find the bugs at build-time and not running-time.
```

File: pages/blog/codegen-this_is_my_type.mdx [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+  tags: ['graphql', 'typescript', 'codegen'],
+  date: '2022-06-08',
+  description: 'GraphQL Code Generator generate TypeScript Discriminated Union Types',
+  image: '/blog-assets/graphql-code-generator/cover.png'
+}
+export default withArticle(meta)
+ 
+# Introduction
+
+In the last few months I just started learn how to code, I already know how to partly work with JavaScript and start to learn TypeScript. When I was create TypeScript types I found a lot of tutorials and documentation that help me create Types and how to implement them in my code. For example: **Discriminated Union.**
```
Comment (kamilkisiela) [inline pages/blog/codegen-this_is_my_type.mdx:14]:
```diff
-When I was create TypeScript
+When I was creating TypeScript
```

Comment (TuvalSimha) [inline pages/blog/codegen-this_is_my_type.mdx:14]:
```suggestion
In the last few months I just started learn how to code, I already know how to partly work with JavaScript and start to learn TypeScript. When I was creating TypeScript types I found a lot of tutorials and documentation that help me create Types and how to implement them in my code. For example: **Discriminated Union.**
```

File: pages/blog/typescript-types.mdx [hunk 8]
```diff
@@ -0,0 +1,159 @@
...
+  date: '2022-07-16',
+  description: 'How to use Typescript basic types system',
+  image: '/blog-assets/graphql-code-generator/cover.png'
+}
+export default withArticle(meta)
+ 
+ 
+# Introduction
+
+TypeScript programming language is a superset of JavaScript that adds types to JavaScript using a set of tools called a type system.
```
Comment (dotansimha) [inline pages/blog/typescript-types.mdx:15]: I think the introduction is too broad - we don't have to talk about TypeScript or what it is, we can assume people who landed on this page know what TypeScript is. 

I would prefer the introduction to cover what's union types and then move to the actual thing.

File: pages/blog/typescript-types.mdx [hunk 9]
```diff
@@ -0,0 +1,159 @@
...
+TypeScript allow us a flexible type that called `any` type like we did in the previous example. But also TypeScript can combine multiple types and combine them together as a union type.
+
+```ts
+let pizzaLover: string | boolean
+
+pizzaLover = 'YES' // string type
+pizzaLover = true // boolean type
+```
+
+More Union type example:
```
Comment (dotansimha) [inline pages/blog/typescript-types.mdx:55]: I thikn the jump from `string | boolean` to `A | B` is fine, but you should also mention here that TS supports string unions as well and give an example. 

```ts
type PizzaAddons = "olives" | "mushrooms" | "pepperoni";

const myPreference: PizzaAddons = "olives"; // ok
const myPreference: PizzaAddons = "pineapple"; // error
```

File: pages/blog/typescript-types.mdx [hunk 10]
```diff
@@ -0,0 +1,159 @@
...
+let pizzaName: string
+pizzaName = 'Napolitana'
+
+pizzaName = 123 // Error: Type number (123) is not assignable to type 'string'.
+```
+
+## Union Types
+
+After we understand how TypeScript type system works, we can start to understand how to use it to create a Union Type.
+TypeScript allow us a flexible type that called `any` type like we did in the previous example. But also TypeScript can combine multiple types and combine them together as a union type.
```
Comment (dotansimha) [inline pages/blog/typescript-types.mdx:46]: Not sure I understand the note about `any`... Also I hate `any` so I don't think it should be mentioned as something you can/should use.

File: pages/blog/typescript-types.mdx [hunk 11]
```diff
@@ -0,0 +1,159 @@
...
+type Pizza = {
+  kind: 'Pizza'
+  topping: string
+  amount: number
+}
+
+type Food = Sushi | Pizza
+```
+
+So what is Type Narrowing?
```
Comment (dotansimha) [inline pages/blog/typescript-types.mdx:73]: Should be `type-narrowing`

File: pages/blog/typescript-types.mdx [hunk 12]
```diff
@@ -0,0 +1,159 @@
...
+```
+
+### Operator `typeof`
+
+Varibale of a union type can assume one or even few diffrent types. In the example below we use `if` statement to check if the variable is of type `Food` or not.
+We can also use the opretor `typeof` with the variable name and compare it with the type you expect for the variable.
+Lets see a basic example of `typeof` operator in function:
+
+```ts
+function getScoreofLovingPizza(score: string | number) {
```
Comment (dotansimha) [inline pages/blog/typescript-types.mdx:108]: I think maybe we can try to provide a more meaningful example here. 
Something like `number | "very-much"` and then translate the "very-much" string into `10` as number?

File: pages/blog/typescript-types.mdx [hunk 13]
```diff
@@ -0,0 +1,159 @@
...
+  kind: 'Pizza'
+  topping: string
+  amount: number
+}
+
+type Food = Sushi | Pizza
+
+function getFoodKindWithFish(Food: Fish | Bird) {
+  if ('fish' in Food) {
+    return Food.fish()
```
Comment (dotansimha) [inline pages/blog/typescript-types.mdx:139]: Not sure I understand this? seems like a syntax error?

**Review by dotansimha — COMMENTED** [review]
Some fixes are needed around the content and code examples.
I also recommend you to use TS playground for all your code examples: https://www.typescriptlang.org/play 
and then you can share a direct link to the live code example.

File: pages/blog/typescript-types.mdx [hunk 14]
```diff
@@ -0,0 +1,153 @@
...
+- I hope this blog post help you better the why of TypeScript types system work and how to use it for your projects.
+
+- In this brief blog post, we looked at a few feature of TypeScript that help you on code-time, mosthly help you to understand the output of any function you create.
+
+- All examples above allows us to build more versatile types by relying on a template pattern for the discriminant property rather than an exact string.
+
+- Recently i beed use GraphQL Codegen plugin to generate my types from my GraphQL schema. When my project is small, as a developer i can create and decalre any type that i want. But when the project is big, it hard to manage the types becuse we have too much types to create.
+  So the best way to manage the types, in my openion, is to use GraphQL Codegen plugin.
+
+## So how do we creating Types based GraphQL schema?
```
Comment (dotansimha) [inline pages/blog/typescript-types.mdx:150]: I think this needs more background:

* GraphQL also has `union` and `interface` -> explain how it connected to the TS feature
* How GraphQL Codegen can help you -> add example

File: pages/blog/typescript-types.mdx [hunk 15]
```diff
@@ -0,0 +1,153 @@
...
+
+## Conclusion
+
+- I hope this blog post help you better the why of TypeScript types system work and how to use it for your projects.
+
+- In this brief blog post, we looked at a few feature of TypeScript that help you on code-time, mosthly help you to understand the output of any function you create.
+
+- All examples above allows us to build more versatile types by relying on a template pattern for the discriminant property rather than an exact string.
+
+- Recently i beed use GraphQL Codegen plugin to generate my types from my GraphQL schema. When my project is small, as a developer i can create and decalre any type that i want. But when the project is big, it hard to manage the types becuse we have too much types to create.
```
Comment (dotansimha) [inline pages/blog/typescript-types.mdx:147]: been?

File: pages/blog/typescript-types.mdx [hunk 16]
```diff
@@ -0,0 +1,145 @@
+import withArticle from '../../ui/blog/article'
+export const meta = {
+  title: 'Back to front: TypeScript union types and how to use GraphQL Codegen to create type-safe',
```
Comment (kamilkisiela) [inline pages/blog/typescript-types.mdx:3]: Hmm I'm not sure I understand the title

**Comment by B2o5T** [comment]
@dotansimha, @charlypoly let's merge this blog post after the new website will out?

File: pages/blog/typescript-types.mdx [hunk 17]
```diff
@@ -0,0 +1,145 @@
...
+  author: 'tuvalSimha',
+  tags: ['graphql', 'typescript', 'codegen'],
+  date: '2022-07-20',
+  description: 'TypeScript union types in use GraphQL schema, and how to use GraphQL Codegen to create type-safe.',
+  image: '/blog-assets/graphql-code-generator/cover.png'
```
Comment (kamilkisiela) [inline pages/blog/typescript-types.mdx:8]: Let's use a cover image with the article's title in it to make it more visible on the website and when shared on social media

File: pages/blog/typescript-types.mdx [hunk 18]
```diff
@@ -0,0 +1,145 @@
...
+}
+export default withArticle(meta)
+ 
+# Introduction
+
+The combination between TypeScript and GraphQL is great. Both TypeScript and GraphQL support the concept of discriminated unions.
+In this short blog post, I want to share my experience in my last project and share one of the most powerful methods to create type-safe on code-time using TypeScript and GraphQL.
```
Comment (kamilkisiela) [inline pages/blog/typescript-types.mdx:15]:
```suggestion
In this short blog post, I want to share my last project experience and share one of the most powerful methods to bring type safety to your code using TypeScript and GraphQL.
```

Maybe something like this?

File: pages/blog/typescript-types.mdx [hunk 19]
```diff
@@ -0,0 +1,145 @@
...
+  name: String!
+  fish: String
+}
+
+type Query {
+  menu: [Menu!]
+}
+```
+
+After we declare our types or create a union type to return some of the others' types, we can generate our types from the schema and use them in the front.
```
Comment (kamilkisiela) [inline pages/blog/typescript-types.mdx:102]:
```suggestion
After we declare our types or create a union type to return some of the others' types, we can generate our types from the schema and use them in the frontend.
```

File: pages/blog/typescript-types.mdx [hunk 20]
```diff
@@ -0,0 +1,145 @@
...
+
+# How to use GraphQL Codegen to generate code and deal with TypeScript union types
+
+First of all, you are more than welcome to explore all the plugins of GraphQL codegen. [GraphQL Code-Generator - LINK](https://graphql-code-generator.com/)
+Like we did in the previous example, let's talk about how to get the data typed safe in the front.
```
Comment (kamilkisiela) [inline pages/blog/typescript-types.mdx:107]:
```suggestion
Like we did in the previous example, let's talk about how to get the data typed safe in the frontend.
```

**Review by kamilkisiela — APPROVED** [review]

File: pages/blog/typescript-types.mdx [hunk 21]
```diff
@@ -0,0 +1,145 @@
...
+  query GetFullMenu {
+    menu {
+      ...Sushi
+      ...Pizza
+    }
+  }
+`
+```
+
+Get the `Menu` type from the schema with the query, so we will be able to get the objects inside the `Menu`.
```
Comment (charlypoly) [inline pages/blog/typescript-types.mdx:124]:
```suggestion
After running codegen using the [`@graphql-codegen/typescript-graphql-request`](https://npmjs.com/package/@graphql-codegen/typescript-graphql-request) plugin, we end up having our result variable (`menu`) properly typed with an union type:
```

**Review by charlypoly — CHANGES_REQUESTED** [review]

File: pages/blog/typescript-types.mdx (hunk 16 above)
Comment (TuvalSimha) [reply pages/blog/typescript-types.mdx]: Charly suggested a new title:

`Getting the best of TypeScript and GraphQL: union types`

**Comment by charlypoly** [comment]
@B2o5T 
> @dotansimha, @charlypoly let's merge this blog post after the new website will out?


@TuvalSimha has other articles pending, @kamilkisiela has one too and I also have to write an article soon so I guess we can move forward with this one :) 

We'll have plenty new articles to showcase our new blog ;)

File: pages/blog/typescript-types.mdx [hunk 22]
```diff
@@ -0,0 +1,145 @@
...
+  tags: ['graphql', 'typescript', 'codegen'],
+  date: '2022-07-20',
+  description: '',
+  image: '/blog-assets/graphql-code-generator/cover.png'
+}
+export default withArticle(meta)
+ 
+# Introduction
+
+The combination between TypeScript and GraphQL is great. Both TypeScript and GraphQL support the concept of discriminated unions.
```
Comment (Urigo) [inline pages/blog/typescript-types.mdx:14]:
```suggestion
The combination between TypeScript and GraphQL can be very powerful. Both TypeScript and GraphQL support the concept of discriminated unions.
```

File: pages/blog/typescript-types.mdx [hunk 23]
```diff
@@ -0,0 +1,145 @@
...
+In this short blog post, I want to share my last project experience and share one of the most powerful methods to create stronger types using TypeScript and GraphQL.
+
+One of my assumptions in this blog post is that you know some TypeScript. If you don't, you can check out ["Basic Types"](https://www.typescriptlang.org/docs/handbook/basic-types.html) handbook.
+Surprisingly, if you have some experience in TypeScript or not, you can understand the Types concept very fast. So, let's talk a little bit about TypeScript union types, one of the most remarkable features of this language.
```
Comment (Urigo) [inline pages/blog/typescript-types.mdx:18]: I'm not sure I understand the initial part of the sentence.
Are you surprised that whether you have the experience or not, you can understand the Types concept fast?

File: pages/blog/typescript-types.mdx [hunk 24]
```diff
@@ -0,0 +1,145 @@
...
+pizzaLover = 'YES' // string type
+pizzaLover = true // boolean type
+```
+
+[Here's a live example in TS Playground](https://www.typescriptlang.org/play?#code/DYUwLgBADglgXnAhgGQPYDcQCcBcEDOYWMAdgOYQA+EARqqqIiQLABQbsCKG2EAvBADkATQCiAZUEQA9NIJFSFMAE8oIDvCRpMWfhCIBXEDLl0GIJvtXrWsiAD0A-GyA)
+
+# How to use union types in TypeScript
+
+TypeScript can also be used to combine complex types such as interfaces.
+You can think about one type that contains some others types. Here simple example that combined 2 types: `Sushi` and `Pizza`, both of them can be united into one type by the name `Food`.
```
Comment (Urigo) [inline pages/blog/typescript-types.mdx:39]:
```suggestion
You can think about one type that contains some others types. Here is a simple example that combines 2 types: `Sushi` and `Pizza`, both of them can be united into one type by the name `Food`.
```

File: pages/blog/typescript-types.mdx [hunk 25]
```diff
@@ -0,0 +1,145 @@
...
+  topping: string
+  amount: number
+}
+
+type Food = Sushi | Pizza
+```
+
+[Here's a live example in TS Playground](https://www.typescriptlang.org/play?#code/C4TwDgpgBAygrgZwBYEsoF4oG8CwAoKKAaxQDsATALigHJ5kUb9CAzFZahYAJzIHNmUAIYBbAPZxSwaqTgiARhG74AvvnyhIUAAooAXnqEZsgkhWo1dBoUwJRgYsGH6ce-QaIlSZcxcrxqeOp4mtAAYmJi5Mb0qFAAPjr6hsFAA)
+
+More Union type example with `string` and `string`:
```
Comment (Urigo) [inline pages/blog/typescript-types.mdx:59]:
```suggestion
And another Union type example, this time with `string` and `string`:
```

File: pages/blog/typescript-types.mdx [hunk 26]
```diff
@@ -0,0 +1,145 @@
...
+
+const myPreference: PizzaAddons = 'olives' // ok
+const myPreference: PizzaAddons = 'pineapple' // error
+```
+
+[Here's a live example in TS Playground](https://www.typescriptlang.org/play?#code/C4TwDgpgBACglgLwQQwIIBN0HsB2BnKAXigHIsAbOANwjxKgB9SBbAVzwAsAnLLZuxqUhhIPHHBIBYAFAyAxrjzAozEDC4QAZhA045EAFyxEKDNnxFSFarXoB6O1CwBreYuWr1WnRD2HjSGiYipYkYHA4EMgi5BD2jjo8XDJAA)
+
+# GraphQL union and TypeScript union type
+
+GraphQL is a query language that is used to define the API of a server. Like TypeScript or even Go language, GraphQL supports union and interface types. The type system in its core principles, so we can use it to make type-safe API calls. The goal is to use it to propagate our backend types to the frontend.
```
Comment (Urigo) [inline pages/blog/typescript-types.mdx:72]:
```suggestion
GraphQL is a query language that is used to define the API of a server. Like TypeScript or even Go language, GraphQL supports union and interface types. The type system is one of its core principles and we can use it to make type-safe API calls. The goal is to use it to propagate our backend types to the frontend.
```

File: pages/blog/typescript-types.mdx [hunk 27]
```diff
@@ -0,0 +1,145 @@
...
+When I just start to learn TypeScript, I found myself did a lot of types errors and annoying misspellings. Those error amounts became bigger and bigger when the project grow.
```
Comment (Urigo) [inline pages/blog/typescript-types.mdx:73]:
```suggestion
When I started to learn TypeScript, I found myself making lots of type errors and annoying misspellings. These error became more frequent as the project grew.
```

File: pages/blog/typescript-types.mdx [hunk 28]
```diff
@@ -0,0 +1,145 @@
...
+The best solution for this situation is to add a type-safety base GraphQL schema.
```
Comment (Urigo) [inline pages/blog/typescript-types.mdx:74]:
```suggestion
The best solution for that situation was to add type-safety based on a GraphQL schema.
```

File: pages/blog/typescript-types.mdx [hunk 29]
```diff
@@ -0,0 +1,145 @@
...
+
+GraphQL schema can return one of the multiple object types when we create Union or Interface. Like TypeScript, we can declare which object types are included in the union:
```
Comment (Urigo) [inline pages/blog/typescript-types.mdx:76]:
```suggestion
A GraphQL schema can return one of multiple object types thanks to the Union or Interface keywords. Like TypeScript, we can declare which object types are included in the union:
```

File: pages/blog/typescript-types.mdx [hunk 30]
```diff
@@ -0,0 +1,145 @@
...
+  name: String!
+  fish: String
+}
+
+type Query {
+  menu: [Menu!]
+}
+```
+
+After we declare our types or create a union type to return some of the others' types, we can generate our types from the schema and use them in the frontend.
```
Comment (Urigo) [inline pages/blog/typescript-types.mdx:102]:
```suggestion
After we declare our types in GraphQL and create a union type to return some of the other types, we can generate TypeScript types from the GraphQL schema and use them in the frontend.
```

File: pages/blog/typescript-types.mdx [hunk 31]
```diff
@@ -0,0 +1,145 @@
...
+
+# How to use GraphQL Codegen to generate code and deal with TypeScript union types
+
+First of all, you are more than welcome to explore all the plugins of GraphQL codegen. [GraphQL Code-Generator - LINK](https://graphql-code-generator.com/)
+Like we did in the previous example, let's talk about getting the data typed safely in the front.
+
+(I use the `react-query` example)
```
Comment (Urigo) [inline pages/blog/typescript-types.mdx:109]:
```suggestion
(in this case, I'll use the `react-query` example)
```

File: pages/blog/typescript-types.mdx [hunk 32]
```diff
@@ -0,0 +1,145 @@
...
+  const { data } = useQuery<MenuQuery>('menu', async () => {
+    const { menu } = await request(endpoint, menuQuery)
+    return menu
+  })
+}
+```
+
+## Conclusion
+
+- I hope this blog post helps you better the why TypeScript unions and GraphQL unions and the way to use them in your project.
```
Comment (Urigo) [inline pages/blog/typescript-types.mdx:137]:
```suggestion
- I hope this blog post helped you to better understand TypeScript unions and GraphQL unions and the way to use them in your project.
```

File: pages/blog/typescript-types.mdx [hunk 33]
```diff
@@ -0,0 +1,145 @@
...
+
+- We can create Union type and Interface in TypeScript and GraphQL that help us to define the API of a server and use it in our client.
```
Comment (Urigo) [inline pages/blog/typescript-types.mdx:139]:
```suggestion
- We can create Union and Interface types in both TypeScript and GraphQL, that helps us to better define the API of a server and use it in our client.
```

File: pages/blog/typescript-types.mdx [hunk 34]
```diff
@@ -0,0 +1,145 @@
...
+
+- A static type check can help safeguard your application. Especially when you building large-scale applications.
```
Comment (Urigo) [inline pages/blog/typescript-types.mdx:141]:
```suggestion
- A static type check can help safeguard your application. Especially in large-scale applications.
```

File: pages/blog/typescript-types.mdx [hunk 35]
```diff
@@ -0,0 +1,145 @@
...
+
+- When you use type-safe GraphQL, it helps you to eliminate existing errors like misspellings and errors, especially in code-time.
+
+- In this brief blog post, we looked at a few features of TypeScript that help you on code-time, most help you to understand the output of any function you create.
```
Comment (Urigo) [inline pages/blog/typescript-types.mdx:145]: can we have a better name for `code-time`? not sure what...

**Review by Urigo — CHANGES_REQUESTED** [review]

**Comment by charlypoly** [comment]
@TuvalSimha, I made the following image for your blog post, wdyt?


The Anatomy of a GraphQL Request-1000x400 (2)

**Comment by TuvalSimha** [comment]
> @TuvalSimha, I made the following image for your blog post, wdyt?
> 
> The Anatomy of a GraphQL Request-1000x400 (2)
> 
> caching

@kamilkisiela

File: pages/blog/typescript-types.mdx [hunk 36]
```diff
@@ -0,0 +1,145 @@
...
+  tags: ['graphql', 'typescript', 'codegen'],
+  date: '2022-07-20',
+  description: '',
+  image: '/blog-assets/graphql-code-generator/cover.png'
+}
+export default withArticle(meta)
+ 
+# Introduction
+
+The combination between TypeScript and GraphQL can be very powerful. Both TypeScript and GraphQL support the concept of discriminated unions.
```
Comment (charlypoly) [inline pages/blog/typescript-types.mdx:14]:
```suggestion
The combination of TypeScript and GraphQL can be very powerful. Both TypeScript and GraphQL support the concept of discriminated unions.
```

**Review by charlypoly — APPROVED** [review]

**Comment by charlypoly** [comment]
@TuvalSimha, I noticed that the previous image I sent you has some vertical alignment issues; here is an updated version:

The Anatomy of a GraphQL Request-1000x400 (4)

**Comment by TuvalSimha** [comment]
Waiting for new website

**Comment by charlypoly** [comment]
@TuvalSimha, I've put the proper images and changed the slug for better SEO

File: pages/blog/typescript-graphql-union-types.mdx [hunk 37]
```diff
@@ -0,0 +1,146 @@
...
+
+First, let's create our `GetFullMenu` query as follows:
+
+(in this case, I'll use the `react-query` example)
+
+```ts
+const menuQuery = gql`
+  query GetFullMenu {
+    menu {
+      ...Sushi
```
Comment (dotansimha) [inline pages/blog/typescript-graphql-union-types.mdx:118]: Add the actual fragment here as part of the code example, it's not clear that you have using specific fragment on each type.

File: pages/blog/typescript-graphql-union-types.mdx [hunk 38]
```diff
@@ -0,0 +1,146 @@
...
+  }
+`
+```
+
+Get the `Menu` type from the schema with the query, so we will be able to get the objects inside the `Menu`.
+
+```ts
+const Menu = () => {
+  const { data } = useQuery<MenuQuery>('menu', async () => {
+    const { menu } = await request(endpoint, menuQuery)
```
Comment (dotansimha) [inline pages/blog/typescript-graphql-union-types.mdx:130]: This example is not 100% clear. Also, I think you should add an example for discriminating the result based on the `__typename` (either using `if` of `switch`)

**Review by dotansimha — CHANGES_REQUESTED** [review]

**Comment by dotansimha** [comment]
> Waiting for new website

I don't think we have to wait for new websites. Why is it needed?

**Comment by dotansimha** [comment]
Also @TuvalSimha please rebase, you have a conflict

File: pages/blog/typescript-graphql-union-types.mdx (hunk 37 above)
Comment (TuvalSimha) [reply pages/blog/typescript-graphql-union-types.mdx]:
```ts
fragment Sushi  {
    name
    fish
  },
fragment Pizza  {
    title
    topping
  }
const menuQuery = gql`
  query GetFullMenu {
    menu {
      ...Sushi
      ...Pizza
    }
  }
`
```

File: packages/guild-website/pages/blog/typescript-types.mdx [hunk 39]
```diff
@@ -0,0 +1,153 @@
...
+  title: 'Getting the best of TypeScript and GraphQL: union types',
+  author: 'tuvalSimha',
+  tags: ['graphql', 'typescript', 'codegen'],
+  date: '2022-07-20',
+  description: '',
+  image: '/blog-assets/typescript-types/typescript-tips.png'
+}
+export default withArticle(meta)
+ 
+# Introduction
```
Comment (B2o5T) [inline packages/guild-website/pages/blog/typescript-types.mdx:12]:
```suggestion
## Introduction
```

File: packages/guild-website/pages/blog/typescript-types.mdx [hunk 40]
```diff
@@ -0,0 +1,153 @@
...
+
+The combination of TypeScript and GraphQL can be very powerful. Both TypeScript and GraphQL support the concept of discriminated unions.
+In this short blog post, I want to share my last project experience and share one of the most powerful methods to create stronger types using TypeScript and GraphQL.
+
+One of my assumptions in this blog post is that you know some TypeScript. If you don't, you can check out ["Basic Types"](https://www.typescriptlang.org/docs/handbook/basic-types.html) handbook.
+So, let's talk a little bit about TypeScript union types, one of the most remarkable features of this language.
+
+## What is a union type in TypeScript?
```
Comment (B2o5T) [inline packages/guild-website/pages/blog/typescript-types.mdx:20]:
```suggestion
### What is a union type in TypeScript?
```

File: packages/guild-website/pages/blog/typescript-types.mdx [hunk 41]
```diff
@@ -0,0 +1,153 @@
...
+```ts
+let pizzaLover: string | boolean // pizzaLover can be a string or boolean type
+
+pizzaLover = 'YES' // string type
+pizzaLover = true // boolean type
+```
+
+[Here's a live example in TS Playground](https://www.typescriptlang.org/play?#code/DYUwLgBADglgXnAhgGQPYDcQCcBcEDOYWMAdgOYQA+EARqqqIiQLABQbsCKG2EAvBADkATQCiAZUEQA9NIJFSFMAE8oIDvCRpMWfhCIBXEDLl0GIJvtXrWsiAD0A-GyA)
+
+## How to use union types in TypeScript
```
Comment (B2o5T) [inline packages/guild-website/pages/blog/typescript-types.mdx:36]:
```suggestion
### How to use union types in TypeScript
```

File: packages/guild-website/pages/blog/typescript-types.mdx [hunk 42]
```diff
@@ -0,0 +1,153 @@
...
+}
+
+type Query {
+  menu: [Menu!]
+}
+```
+
+After we declare our types in GraphQL and create a union type to return some of the other types, we can generate TypeScript types from the GraphQL schema and use them in the frontend.
+
+## How to use GraphQL Codegen to generate code and deal with TypeScript union types
```
Comment (B2o5T) [inline packages/guild-website/pages/blog/typescript-types.mdx:104]:
```suggestion
### How to use GraphQL Codegen to generate code and deal with TypeScript union types
```

**Comment by B2o5T** [comment]
can we choose the more specific page name for blog post? current is `typescript-types.mdx`

**Comment by charlypoly** [comment]
@TuvalSimha I'll publish this one today 🚀

**Comment by TuvalSimha** [comment]
> @TuvalSimha I'll publish this one today 🚀

Thank you!!!

**Pull request merged**